### PR TITLE
[improvement][Undertow] Upgrade the undertow version required by conjure-undertow

### PR DESCRIPTION
**Pull request by carterkozak**

### CODE REVIEW

File: conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureExceptionHandler.java
Comment (rubenfiszel) [inline conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureExceptionHandler.java:139]: nice

File: conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/Undertow1460Handler.java
Comment (dansanduleac) [inline conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/Undertow1460Handler.java:23]: guess you meant 2.0.17.Final!

File: conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureUrlDecodingHandler.java
Comment (rubenfiszel) [inline conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureUrlDecodingHandler.java:100]: how come the exchange can still be in the IoThread ? I thought handlers were executed by the worker thread

File: conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/Undertow1460Handler.java
Comment (carterkozak) [reply conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/Undertow1460Handler.java:23]: Doh, woops!

File: conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureUrlDecodingHandler.java
Comment (rubenfiszel) [inline conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureUrlDecodingHandler.java]: newValues ?

File: conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureUrlDecodingHandler.java
Comment (carterkozak) [reply conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureUrlDecodingHandler.java:100]: Requests are received on an IO thread, once they reach BlockingHandler (which we register in ConjureHandler) they will transition to a worker thread.

We're generally best off doing as much non-blocking initialization on the IO thread since they're meant to be constantly available and scale with CPU, which has a side-effect of reducing the stack depth on the task thread (where service methods are executed) for prettier stack traces.

File: conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureUrlDecodingHandler.java
Comment (carterkozak) [reply conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureUrlDecodingHandler.java]: Nice catch, updated

File: conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureUrlDecodingHandler.java
Comment (rubenfiszel) [inline conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureUrlDecodingHandler.java:67]: Wondering if it would help to have a variable holding the lambda,
 `Function1<String, String> fun = str -> URLUtils.decode(str, charset, decodeSlash, false, sb)` to remove all the redundancy

**Review by rubenfiszel — APPROVED** [review]

File: conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureUrlDecodingHandler.java
Comment (iamdanfox) [inline conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureUrlDecodingHandler.java:35]: It feels a tiny bit funky to be backporting unmerged PRs - how urgent do you think the form encoding bugfix actually is - would it be tolerable to just stay on vanilla undertow for a bit longer until that PR merges?

**Review by iamdanfox — APPROVED** [review]
Upgrade looks fine, defer to you guys whether you're happy to take the proposed ConjureUrlDecodingHandler before upstream merges!

File: conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureUrlDecodingHandler.java
Comment (rubenfiszel) [reply conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureUrlDecodingHandler.java:100]: `We're generally best off doing as much non-blocking initialization on the IO thread since they're meant to be constantly available and scale with CPU` Is that always true ? Isn't there a point where ioThreads become the new bottleneck and reduce the possibility of doing some service reprioritization (which ofc we don't do, but just in theory) ?

**Pull request merged**
